float32 x_coord
float32 y_coord
int32 face_id
float32 rr_x
float32 rr_y
float32 rr_z
float32 rr_w
float32 normal_x
float32 normal_y
float32 face_c_x
float32 face_c_y
string identity